internal/test/name.go

```diff
@@ -0,0 +1,47 @@
+package test
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Name returns the test name based on the provided fields and values.
+func Name(fields []string, values ...any) string {
+	if len(fields) != len(values) {
+		panic("fields and values must have the same length")
+	}
+	b := strings.Builder{}
+	sep := ""
+	for i, f := range fields {
+		v := values[i]
```

Comment:
We should probably discuss whether or not to print the zero value or drop it. Maybe it would make sense to have a trick to encode "print even if zero-value".